components/shared/embedder/lib.rs

```diff
@@ -433,3 +437,114 @@ pub struct DualRumbleEffectParams {
 pub enum GamepadHapticEffectType {
     DualRumble(DualRumbleEffectParams),
 }
+
+#[derive(Clone, Debug, Deserialize, MallocSizeOf, Serialize)]
+pub struct WebResourceRequest {
+    #[serde(
+        deserialize_with = "::hyper_serde::deserialize",
+        serialize_with = "::hyper_serde::serialize"
+    )]
+    #[ignore_malloc_size_of = "Defined in hyper"]
+    pub method: Method,
+    pub url: ServoUrl,
+    #[serde(
+        deserialize_with = "::hyper_serde::deserialize",
+        serialize_with = "::hyper_serde::serialize"
+    )]
+    #[ignore_malloc_size_of = "Defined in hyper"]
+    pub headers: HeaderMap,
+    pub is_redirect: bool,
+    pub is_main_frame: bool,
```

Comment:
Why these two properties in particular?

Reply:
I would like to align with these APIs :[Android WebResourceRequest](https://developer.android.com/reference/android/webkit/WebResourceRequest?_gl=1*1u2thnp*_up*MQ..*_ga*MTY3NTEzMDYzNi4xNzM2OTE2ODcx*_ga_6HH9YJMN9M*MTczNjkxNjg3MS4xLjAuMTczNjkxNjg3MS4wLjAuMTc4ODEwMjI5MQ..) [GeckoView's LoadRequest](https://mozilla.github.io/geckoview/javadoc/mozilla-central/org/mozilla/geckoview/GeckoSession.NavigationDelegate.LoadRequest.html). I think these APIs are widely used, making it easier for developers to transition seamlessly to the Servo embedder API.